src/elliptic/curves/wrappers/point.rs

```diff
@@ -178,23 +180,68 @@ impl<E: Curve> Point<E> {
         }
     }
 
+    /// Constructs a `&Point<E>` from reference to low-level [ECPoint] implementation
+    ///
+    /// Returns error if point is not valid. Valid point is either a zero point, or a point of
+    /// [group order].
+    ///
+    /// Typically, you don't need to use this constructor. See [generator](Point::generator),
+    /// [base_point2](Point::base_point2) constructors.
+    ///
+    /// [ECPoint]: crate::elliptic::curves::ECPoint
+    /// [group order]: crate::elliptic::curves::ECScalar::group_order
+    pub fn from_raw_ref(raw_point: &E::Point) -> Result<&Self, MismatchedPointOrder> {
+        if raw_point.is_zero() || raw_point.check_point_order_equals_group_order() {
+            // Safety: Self is repr(transparent) wrapper over E::Point => cast is sound
+            let reference = unsafe { &*(raw_point as *const E::Point as *const Self) };
```

Comment:
nit, Call `from_raw_ref_unchecked()` and change the SAFETY comment here to say that we checked that the point is either zero or the group order and hence satisfies the SAFETY invariant of `from_raw_ref_unchecked`

Reply:
(that way the actual unsafe code is only in a single function)

Reply:
You're right, good point!